zipkin-storage/elasticsearch/src/test/java/zipkin2/elasticsearch/integration/ITEnsureIndexTemplate.java

```diff
@@ -0,0 +1,126 @@
+   */
+  protected abstract ElasticsearchStorage.Builder newStorageBuilder(TestInfo testInfo);
+
+  @BeforeAll void setUpCatchAllTemplate(TestInfo testInfo) throws IOException {
+    ElasticsearchStorage.Builder builder = newStorageBuilder(testInfo);
+    storage = builder.build();
+    assumeTrue(storage.version() >= 7.8f, () -> "Applicable only for ES >= 7.8, skipping test");
+
+    /**
+     * Delete all index templates in order to create the "catch-all" index template, because
+     * ES does not allow multiple index templates of the same index_patterns and priority
+     */
+    deleteIndexTemplate("*");
+    setUpCatchAllTemplate();
+  }
+
+  @Test void createZipkinIndexTemplate_getTraces_returnsSuccess() throws IOException {
+    /** Create index template with {@link ElasticsearchStorage#ensureIndexTemplates()} */
+    CheckResult check = storage.check();
+
+    assertThat(check.ok()).isTrue();
+
+    List<String> traceIds = asList(LOTS_OF_SPANS[0].traceId(), LOTS_OF_SPANS[1].traceId());
+    List<Span> spans = asList(LOTS_OF_SPANS[0], LOTS_OF_SPANS[1]);
+
+    storage.spanConsumer().accept(spans).execute();
+
+    assertThat(storage.traces().getTraces(traceIds).execute())
+      .containsOnly(asList(LOTS_OF_SPANS[0]), asList(LOTS_OF_SPANS[1]));
+  }
+
+  /**
+   * Create a "catch-all" index template with the lowest priority prior to running tests to
+   * ensure that the index templates created during tests with higher priority function as
+   * designed. Only applicable for ES >= 7.8
+   */
+  void setUpCatchAllTemplate() throws IOException {
+    AggregatedHttpRequest updateTemplate = AggregatedHttpRequest.of(
+      RequestHeaders.of(
+        HttpMethod.PUT, catchAllIndexPath(), HttpHeaderNames.CONTENT_TYPE, MediaType.JSON_UTF_8),
+      HttpData.ofUtf8(catchAllTemplate()));
+    Internal.instance.http(storage).newCall(updateTemplate, (parser, contentString) -> null,
+      "update-template").execute();
+  }
+
+  String catchAllIndexPath() {
+    return "/_index_template/catch-all";
+  }
+
+  String catchAllTemplate() {
+    return "{\n"
+      + "  \"index_patterns\" : [\"*\"],\n"
+      + "  \"priority\" : 0,\n"
+      + "  \"template\": {\n"
+      + "    \"settings\" : {\n"
+      + "      \"number_of_shards\" : 1\n"
+      + "    },\n"
+      + "    \"mappings\" : {\n"
+      + "      \"_source\" : { \"enabled\" : true }\n"
+      + "    }\n"
+      + "  }\n"
+      + "}";
+  }
+
+  void deleteIndexTemplate(String pattern) throws IOException {
+    String url = "/_index_template/" + pattern;
+    AggregatedHttpRequest delete = AggregatedHttpRequest.of(HttpMethod.DELETE, url);
+    Internal.instance.http(storage)
+      .newCall(delete, (parser, contentString) -> null, "delete-index").execute();
+  }
+
+  /**
+   * Delete "catch-all" index template so it does not interfere with any other test
+   */
+  @AfterAll void deleteCatchAllTemplate() throws IOException {
```

Comment:
this should be  a "finally" of the catchAllTemplateTest as it would only need to be tested once anyway